explainers/signed-exchange-alternate-link-explainer.md

```diff
@@ -0,0 +1,164 @@
+1. If exists, check whether matching “alternate” link HTTP header in the outer response exists or not.
+1. If exists, prefetches the matching signed exchange instead of prefetching the original resource URL.
+1. The prefetched signed exchange will be stored to the SignedExchangeCache of the Document. And it will be passed to the next Document and used while processing the preload link header. This behavior is written in [Signed Exchange subresource substitution explainer](./signed-exchange-subresource-subtitution-explainer.md).
+
+## Detailed design discussion
+### Content negotiation using Variants and Variant-Key
+The **alternate** link header and **allowed-alt-sxg** link headers can have [variants and variant-key](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html) attributes to support content negotiation (eg: [WebP support](https://developers.google.com/speed/webp/faq#server-side_content_negotiation_via_accept_headers)).
+
+- In **outer** HTTP response of article.html.sxg from feed.example:
+    ```
+    Link: <https://feed.example/publisher.example/image_jpeg.sxg>;
+          rel="alternate";
+          type="application/signed-exchange;v=b3";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/jpeg";
+          anchor="https://publisher.example/image";
+    Link: <https://feed.example/publisher.example/image_webp.sxg>;
+          rel="alternate";
+          type="application/signed-exchange;v=b3";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/webp";
+          anchor="https://publisher.example/image";
+    ```
+- In **inner** response header of article.html.sxg:
+    ```
+    Link: <https://publisher.example/image>;
+          rel="allowed-alt-sxg";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/jpeg";
+          header-integrity="sha256-AAA"
+    Link: <https://publisher.example/image>;
+          rel="allowed-alt-sxg";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/webp";
+          header-integrity="sha256-BBB"
+    Link: <https://publisher.example/image>; rel=preload; as=image;
+    ```
+If a UA supports WebP, the UA should prefetch **image_webp.sxg** which content is WebP format. Otherwise the UA should prefetch **image_jpeg.sxg** which content is JPEG format.
+
+## Security and Privacy Considerations
+The UA must fetch the alternate signed exchange subresource (lib.js.sxg) even if there is the original subresource (lib.js) in the HTTPCache. Otherwise it leaks the state of publisher’s site to the distributor of the signed exchange.
```

Comment:
Also, to prevent tracking (user ID transfer), if the aggregator failed to prefetch a subresource that the main resource preloads, the UA must drop all of the subresource prefetches. If the aggregator prefetches a superset of the preloaded subresources, the UA must drop the ones that weren't preloaded.

Reply:
Done